neasqc_wp61/models/quantum/embedder.py

```diff
@@ -0,0 +1,322 @@
+"""
+Embedder
+============
+Module containing the base class for transforming sentences in a dataset into embeddings
+"""
+
+from abc import ABC, abstractmethod
+
+import pandas as pd
+from pandas.core.api import DataFrame as DataFrame
+from transformers import BertTokenizer, BertModel
+import torch
+import numpy as np
+import fasttext as ft
+import fasttext.util as ftu
+
+
+class Embedder(ABC):
+    """
+    Base class for the generation of embeddings for the sentences
+    comprising a dataset.
+    """
+
+    def __init__(self, dataset: DataFrame, sentence_or_word: bool = True) -> None:
+        """
+        Initialises the embedder class
+
+        Parameters
+        ----------
+        dataset : pd.DataFrame
+            Pandas dataset. Each row of the dataset correspods to a
+            sentence.
+        sentence_or_word : bool
+            A bool controllig whether we want to produce a sentence
+            embedding or word embedding vector for each sentence in the
+            dataset. True is for sentence embedding, False is for word
+            embedding.
+        """
+
+        self.dataset = dataset
+        self.sentence_or_word = sentence_or_word
+
+    @abstractmethod
+    def compute_embeddings(self) -> DataFrame:
+        """
+        Creates embeddings for sentences in a dataset
+
+        Returns
+        -------
+        pd.DataFrame
+            A pandas dataframe with the same structure as the original
+            dataset, but with an additional column containing the
+            embeddings corresponding to the sentence contained in each
+            row.
+        """
+        pass
```

Comment:
this should raise a not implemented error so you make sure you over-write it in child classes

Reply:
Interesting, I do not get any errors.. Could you tell me how you obtained this error so I can try and replicate it? Thanks!

Reply:
No, it doesn't. But it should! Lilke, this is the behaviour you want from an abstract method. You do nOT want "pass", you want to throw/raise a NotImplementedError. pls google how to do that :).

Reply:
addressed